tony/README.md

```diff
@@ -0,0 +1,53 @@
+# TonY - TensorFlow on YARN
+
+This [initialization action](https://cloud.google.com/dataproc/init-actions) installs the latest version of [TonY](https://github.com/linkedin/TonY)
+on a master node within a [Google Cloud Dataproc](https://cloud.google.com/dataproc) cluster.
+
+## Using this initialization action
+
+You can use this initialization action to create a new Dataproc cluster with TonY installed:
+
+1. Use the `gcloud` command to create a new cluster with this initialization action.  The following command will create a new cluster named `<CLUSTER_NAME>`.
+
+    ```bash
+    gcloud dataproc clusters create <CLUSTER_NAME> \
+      --initialization-actions gs://dataproc-initialization-actions/tony/tony.sh
+    ```
+
+    You can also pass specific metadata:
+    
+    ```bash
+    gcloud dataproc clusters create <CLUSTER_NAME> \
+      --initialization-actions gs://dataproc-initialization-actions/tony/tony.sh \
+      --metadata name1=value1,name2=value2... 
+    ```
+    
+    Supported metadata parameters:
+    
+     - worker_instances 
+     - worker_memory
+     - ps_instances
+     - ps_memory
+        
+    These parameters are defined here: TonY [configurations](https://github.com/linkedin/TonY/wiki/TonY-Configurations)
+    
+    Example:
+    
+    ```bash
+    gcloud dataproc clusters create <CLUSTER_NAME> \
+      --initialization-actions gs://dataproc-initialization-actions/tony/tony.sh \
+      --metadata worker_instances=4,worker_memory=4g,ps_memory=2g
```

Comment:
Can we make Tony, TensorFlow and other software versions configurable through metadata?